File: 0485-max-consecutive-ones/0485-max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        ans = 0
        newBest = 0
        n = len(nums)
        for i in range(0, n):
        
            if(nums[i] == 1):
            
                newBest+=1
                if(newBest > ans):
                    ans=newBest
            
            else:
                newBest = 0
        

        return ans

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int ans = 0;
        int newBest = 0;
        for(int i=0; i<nums.size(); i++)
        {
            if(nums[i] == 1)
            {
                newBest++;
                if(newBest > ans)
                    ans=newBest;
            }
            else
                newBest = 0;
        }

        return ans;
    }
};